inc/commonglpi.class.php

```diff
@@ -1218,6 +1217,11 @@ function display($options = []) {
       // $options must contains the id of the object, and if locked by manageObjectLock will contains 'locked' => 1
       ObjectLock::manageObjectLock(get_class($this), $options);
 
+      // manage reservation options
+      if (isset($_REQUEST['defaultDate'])) {
+         $options['defaultDate'] = $_REQUEST['defaultDate'];
+      }
+
```

Comment:
This is really specific. I propose to add something more generic that may help for many usecases:

```diff
diff --git a/inc/commonglpi.class.php b/inc/commonglpi.class.php
index bea476505a..8ae387ac1e 100644
--- a/inc/commonglpi.class.php
+++ b/inc/commonglpi.class.php
@@ -1217,9 +1217,9 @@ class CommonGLPI {
       // $options must contains the id of the object, and if locked by manageObjectLock will contains 'locked' => 1
       ObjectLock::manageObjectLock(get_class($this), $options);
 
-      // manage reservation options
-      if (isset($_REQUEST['defaultDate'])) {
-         $options['defaultDate'] = $_REQUEST['defaultDate'];
+      // manage custom options
+      if (isset($_REQUEST['tab_params']) && is_array($_REQUEST['tab_params'])) {
+         $options += $_REQUEST['tab_params'];
       }
 
       $this->showNavigationHeader($options);
diff --git a/inc/html.class.php b/inc/html.class.php
index baf70933b5..6c2685894e 100644
--- a/inc/html.class.php
+++ b/inc/html.class.php
@@ -914,6 +914,7 @@ class Html {
          if (isset($url['query'])) {
             parse_str($url['query'], $parameters);
             unset($parameters['forcetab']);
+            unset($parameters['tab_params']);
             $new_query = http_build_query($parameters);
             return str_replace($url['query'], $new_query, $url_in);
          }
diff --git a/inc/reservation.class.php b/inc/reservation.class.php
index d13bbbca87..653c9de2d9 100644
--- a/inc/reservation.class.php
+++ b/inc/reservation.class.php
@@ -1170,7 +1170,7 @@ JAVASCRIPT;
             echo "<td>";
             list($annee, $mois, $jour) = explode("-", $data["begin"]);
             echo "<a href='".$item::getFormURLWithID($ri->fields['items_id']).
-                 "&forcetab=Reservation$1&defaultDate={$data["begin"]}' ".
+                 "&forcetab=Reservation$1&tab_params[defaultDate]={$data["begin"]}' ".
                   "title=\"".__s('See planning')."\">";
             echo "<i class='far fa-calendar-alt'></i>";
             echo "<span class='sr-only'>".__('See planning')."</span>";
```